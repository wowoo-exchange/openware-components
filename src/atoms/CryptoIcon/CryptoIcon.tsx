import * as React from 'react';

import '@openware/cryptofont';
import cx from 'classnames';

interface CryptoIconProps {
    /**
     * Crypto currency code from
     * https://github.com/atomiclabs/cryptocurrency-icons/blob/master/manifest.json
     * with or without "-alt"
     */
    code: string;

    /**
     * Additional class name for styling. By default element receives `rc-crypto-font` class
     * @default empty
     */
    className?: string;
    /**
     * Anything you wish to attach an icon to
     * @default empty
     */
    children?: React.ReactNode;
}

export const CR_CURRENCY_ICON_CLASS_NAAME = 'cr-crypto-font';

const makeCryptoClass = (code: string) => `${CR_CURRENCY_ICON_CLASS_NAAME}-${code}`;

/**
 * Component to display crypto-currency icon as a standalone (styleable)
 * image or with a text next to it (usually, some float number).
 */
const CryptoIcon: React.FunctionComponent<CryptoIconProps> = props => {
    const { code, className = '', children } = props;
    return (
        <span className={cx('cr-crypto-icon', className)}>
            <span className={makeCryptoClass(code)}> {children}</span>
        </span>
    );
};

export {
    CryptoIcon,
    CryptoIconProps,
};
