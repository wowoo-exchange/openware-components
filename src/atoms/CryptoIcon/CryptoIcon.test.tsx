import { shallow } from 'enzyme';
import * as React from 'react';
import { CryptoIcon, CryptoIconProps } from './CryptoIcon';

const defaultProps: CryptoIconProps = {
    code: 'BTC',
};

const setup = (props: Partial<CryptoIconProps> = {}) =>
    shallow(<CryptoIcon {...{ ...defaultProps, ...props }} />);

describe('CryptoIcon', () => {

    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-crypto-icon')).toBeTruthy();
    });

    it('should render children', () => {
        const amount = 123;
        const wrapper = setup({ children: [amount] });
        expect(wrapper.text()).toEqual(` ${amount}`);
    });

    it('should pull icon with the right icon class', () => {
        const icon = setup().children('span');
        expect(icon.hasClass(`cr-crypto-font-${defaultProps.code}`))
            .toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

});
