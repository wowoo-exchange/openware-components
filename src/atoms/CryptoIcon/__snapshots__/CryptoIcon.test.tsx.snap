// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoIcon should render 1`] = `
<span
  className="cr-crypto-icon"
>
  <span
    className="cr-crypto-font-BTC"
  >
     
  </span>
</span>
`;
